shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -44,6 +44,38 @@
 NSNotificationName const FlutterViewControllerShowHomeIndicator =
     @"FlutterViewControllerShowHomeIndicator";
 
+/**
+ * Compute the interpolated value under linear interpolation.
+ */
+CGFloat FLTLinearInterpolatedValue(double progress, CGFloat from, CGFloat to, CGFloat scale) {
+  NSCAssert(progress >= 0 && progress <= 1, @"progress must be between 0 and 1");
+  return (from * (1 - progress) + to * progress) * scale;
+}
+
+/**
+ * Interpolate the viewport metrics for smoother rotation transition.
+ */
+void FLTInterpolateViewportMetrics(flutter::ViewportMetrics& viewportMetrics,
+                                   double rotationProgress,
+                                   CGSize fromSize,
+                                   UIEdgeInsets fromPadding,
+                                   CGSize toSize,
+                                   UIEdgeInsets toPadding) {
+  CGFloat scale = [UIScreen mainScreen].scale;
```

Comment:
Suggest checking if the view is loaded before doing this:
```suggestion
  UIScreen* mainScreen = self.mainScreenIfViewLoaded;
  if (mainScreen == nil) {
    return;
  }
  CGFloat scale = mainScreen.scale;
```

Reply:
Just did some research - it turned out to be more involved and likely requires non-trivial refactoring to make it work in unit test, as explained [here](https://github.com/flutter/engine/pull/40412/files#diff-365b13daa5168d36d5824f220bc472a4a90f4685761947f8daf9dc76e7f8d962R915-R919).